staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -743,7 +743,7 @@ func (s *Controller) lockedUpdateLoadBalancerHosts(service *v1.Service, hosts []
 }
 
 func wantsLoadBalancer(service *v1.Service) bool {
-	return service.Spec.Type == v1.ServiceTypeLoadBalancer
+	return service.Spec.Type == v1.ServiceTypeLoadBalancer && service.Spec.LoadBalancerClassName == nil
```

Comment:
can we add a unit test for service controller to ignore Services with a valid class name?